server/src/main/java/org/elasticsearch/health/GetHealthAction.java

```diff
@@ -0,0 +1,242 @@
+            builder.field("status", "green");
+            builder.field("cluster_name", clusterName.value());
+            builder.array("impacts");
+            builder.startObject("components");
+            for (Component component : components) {
+                builder.startObject(component.getName());
+                builder.field("status", component.getStatus());
+                builder.startArray("indicators");
+                List<Indicator> indicators = component.getIndicators();
+                for (Indicator indicator : indicators) {
+                    builder.startObject();
+                    builder.field("name", indicator.getName());
+                    builder.field("status", indicator.getStatus());
+                    builder.field("explain", indicator.getExplain());
+                    builder.startObject("meta");
+                    indicator.toXContent(builder, params);
+                    builder.endObject();
+                    // TODO: Add detail / documentation
+                    builder.endObject();
+                }
+                builder.endArray();
+                builder.endObject();
+            }
+            builder.endObject();
+            return builder.endObject();
+        }
+    }
+
+    public static class Request extends ActionRequest {
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+    }
+
+    public static class TransportAction extends org.elasticsearch.action.support.TransportAction<Request, Response> {
+
+        private final ClusterService clusterService;
+
+        @Inject
+        public TransportAction(
+            final ActionFilters actionFilters,
+            final TransportService transportService,
+            final ClusterService clusterService
+        ) {
+            super(NAME, actionFilters, transportService.getTaskManager());
+            this.clusterService = clusterService;
+        }
+
+        @Override
+        protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
+            final ClusterState clusterState = clusterService.state();
+            final Controller component = new Controller(clusterService.localNode(), clusterState);
+            listener.onResponse(new Response(clusterService.getClusterName(), Arrays.asList(component, new Backups())));
+        }
+    }
+
+    private interface Component {
+
+        String getName();
+
+        ClusterHealthStatus getStatus();
+
+        List<Indicator> getIndicators();
+
+    }
+
+    private interface Indicator extends ToXContentFragment {
+
+        String getName();
+
+        ClusterHealthStatus getStatus();
+
+        String getExplain();
+
+    }
+
+    private record NodeDoesNotHaveMaster(DiscoveryNode node, DiscoveryNode masterNode) implements Indicator {
+
+        public static final String GREEN_EXPLAIN = "health coordinating instance has a master node";
```

Comment:
Should the capitalization on these be sentence case? Also, are these user-actionable explanations?

Reply:
I went with capitalization since this is a description. We tend to view this "explain" currently as a user readable explanation of the issue. We will add a documentation link or something like that which is more what we think of as "paragraph" summaries and debugging / resolution steps.